weblate/addons/base.py

```diff
@@ -72,27 +74,46 @@ def get_identifier(cls):
         return cls.name
 
     @classmethod
-    def create_object(cls, component, **kwargs):
+    def create_object(
+        cls,
+        *,
+        component: Component | None = None,
+        project: Project | None = None,
+        **kwargs,
+    ):
         from weblate.addons.models import Addon
 
-        result = Addon(component=component, name=cls.name, **kwargs)
+        result = Addon(project=project, component=component, name=cls.name, **kwargs)
+
         result.addon_class = cls
         return result
 
     @classmethod
-    def create(cls, component, run: bool = True, **kwargs):
-        storage = cls.create_object(component, **kwargs)
+    def create(
+        cls,
+        component: Component | None = None,
+        project: Project | None = None,
+        run: bool = True,
+        **kwargs,
+    ):
+        storage = cls.create_object(component=component, project=project, **kwargs)
         storage.save(force_insert=True)
         result = cls(storage)
         result.post_configure(run=run)
         return result
 
     @classmethod
-    def get_add_form(cls, user, component, **kwargs):
+    def get_add_form(
+        cls,
+        user,
+        component: Component | None = None,
```

Comment:
Make these kwargs only so that it is clear whether component or project is passed:

```suggestion
        *,
        component: Component | None = None,
```